platform/common/emu.c

```diff
@@ -1,1922 +1,1915 @@
-	{
-		int is_stereo = (PicoIn.opt & POPT_EN_STEREO) ? 1 : 0;
-		/// Hard Bypass Stereo to mono
-		//is_stereo = 0;
-
-		PsndRerate(Pico.m.frame_count ? 1 : 0);
-
-		printf("starting audio: %i len: %i stereo: %i, pal: %i\n",
-			PicoIn.sndRate, Pico.snd.len, is_stereo, Pico.m.pal);
-		sndout_start(PicoIn.sndRate, is_stereo);
-		PicoIn.writeSound = snd_write_nonblocking;
-		plat_update_volume(0, 0);
-		memset(sndBuffer, 0, sizeof(sndBuffer));
-		PicoIn.sndOut = sndBuffer;
-	}
-}
-
-void emu_sound_stop(void)
-{
-	sndout_stop();
-}
-
-void emu_sound_wait(void)
-{
-	sndout_wait();
-}
-
-static void emu_loop_prep(void)
-{
-	static int pal_old = -1;
-	static int filter_old = -1;
-
-	if (currentConfig.CPUclock != plat_target_cpu_clock_get())
-		plat_target_cpu_clock_set(currentConfig.CPUclock);
-
-	if (Pico.m.pal != pal_old) {
-		plat_target_lcdrate_set(Pico.m.pal);
-		pal_old = Pico.m.pal;
-	}
-
-	if (currentConfig.filter != filter_old) {
-		plat_target_hwfilter_set(currentConfig.filter);
-		filter_old = currentConfig.filter;
-	}
-
-	plat_target_gamma_set(currentConfig.gamma, 0);
-
-	pemu_loop_prep();
-}
-
-/* our tick here is 1 us right now */
-#define ms_to_ticks(x) (unsigned int)(x * 1000)
-#define get_ticks() plat_get_ticks_us()
-
-void emu_loop(void)
-{
-	int frames_done, frames_shown;	/* actual frames for fps counter */
-	int frame_nb = 0;
-	int target_frametime_x3;
-	unsigned int timestamp_x3 = 0;
-	unsigned int timestamp_aim_x3 = 0;
-	unsigned int timestamp_fps_x3 = 0;
-	char *notice_msg = NULL;
-	char fpsbuff[24];
-	int fskip_cnt = 0;
-
-	fpsbuff[0] = 0;
-
-	PicoLoopPrepare();
-
-	plat_video_loop_prepare();
-	emu_loop_prep();
-	pemu_sound_start();
-
-	/* number of ticks per frame */
-	if (Pico.m.pal)
-		target_frametime_x3 = 3 * ms_to_ticks(1000) / 50;
-	else
-		target_frametime_x3 = 3 * ms_to_ticks(1000) / 60;
-
-	reset_timing = 1;
-	frames_done = frames_shown = 0;
-
-	/* loop with resync every 1 sec. */
-	while (engineState == PGS_Running)
-	{
-		int skip = 0;
-		int diff;
-
-		pprof_start(main);
-
-		if (reset_timing) {
-			reset_timing = 0;
-			plat_video_wait_vsync();
-			timestamp_aim_x3 = get_ticks() * 3;
-			timestamp_fps_x3 = timestamp_aim_x3;
-			fskip_cnt = 0;
-		}
-		else if (currentConfig.EmuOpt & EOPT_NO_FRMLIMIT) {
-			timestamp_aim_x3 = get_ticks() * 3;
-		}
-
-		timestamp_x3 = get_ticks() * 3;
-
-		// show notice_msg message?
-		if (notice_msg_time != 0)
-		{
-			static int noticeMsgSum;
-			if (timestamp_x3 - ms_to_ticks(notice_msg_time) * 3
-			     > ms_to_ticks(STATUS_MSG_TIMEOUT) * 3)
-			{
-				notice_msg_time = 0;
-				notice_msg = NULL;
-				plat_status_msg_clear();
-			}
-			else {
-				int sum = noticeMsg[0] + noticeMsg[1] + noticeMsg[2];
-				if (sum != noticeMsgSum) {
-					plat_status_msg_clear();
-					noticeMsgSum = sum;
-				}
-				notice_msg = noticeMsg;
-			}
-		}
-
-		// second changed?
-		if (timestamp_x3 - timestamp_fps_x3 >= ms_to_ticks(1000) * 3)
-		{
-
-#define FUNKEY_RESYNCHRONIZE_AUDIO_SECS	(5*60)
```

Comment:
I removed this hack as irixxxx helped me fixing the audio desync here: https://github.com/DrUm78/picofe-irixxxx/commit/e87e15f1702e4182ebabb1ab9df1fb9ff13a2e95 (I forked your libpicofe).
Please note that the file is now in CR/LF as notaz/irixxxx's file uses those EOL.